### DISCUSSION TITLE
Do not send request to glyphs and not working with plugin: "maplibre-gl-measures"

### DISCUSSION THREAD

**Discussion by SO-Matveev** [jdsantos/maplibre-gl-measures#20] [Q&A] [answered]
I have demo project from Angular. I create one page and one components map-layout for maplibre-gl-js. Then i install plugin maplibre-gl-measures. This is my code from example
````
 import MeasuresControl from 'maplibre-gl-measures'

@Component({
  selector: "app-map-layout",
  templateUrl: "./map-layout.component.html",
  standalone: true,

  styleUrls: ["./map-layout.component.css"]
})

export class MapLayoutComponent implements OnInit{
  ngOnInit(): void {
    const map: maplibregl.Map = new maplibregl.Map({
      container: 'map', // container id
      style: {
        version: 8,
        sources: {
          osm: {
            type: "raster",
            tiles: ["https://a.tile.openstreetmap.org/{z}/{x}/{y}.png"],
            tileSize: 256,
            attribution: "&copy; OpenStreetMap Contributors",
            maxzoom: 19
          }
        },
        glyphs: "https://fonts.openmaptiles.org/{fontstack}/{range}.pbf",
        layers: [
          {
            id: "osm",
            type: "raster",
            source: "osm"
          }
        ]
      },
      center: [-8, 40], // starting position [lng, lat]
      zoom: 15 // starting zoom
    });
    console.log('map', map)
    map.addControl(new MeasuresControl(), "top-left");

  }
}
````
Then i added new control from maplibre-gl-measures and he partially works.
I see button, he paint lines or polygon but i not see total area or length.
![map](https://github.com/user-attachments/assets/40e41a31-d1ea-47a5-ac2b-123514410d16)

The mistake is that maplibre-gl does not send request to glyphs at the specified address as in the example.

Please, help!
What am I doing wrong?

I tried to use local glyphs. Also not worked.
Unfortunately maplibre-gl not used glyphs, but when open example it`s working

**Comment by jdsantos** [comment]
> I have demo project from Angular. I create one page and one components map-layout for maplibre-gl-js. Then i install plugin maplibre-gl-measures. This is my code from example
> ````
>  import MeasuresControl from 'maplibre-gl-measures'
> 
> @Component({
>   selector: "app-map-layout",
>   templateUrl: "./map-layout.component.html",
>   standalone: true,
> 
>   styleUrls: ["./map-layout.component.css"]
> })
> 
> export class MapLayoutComponent implements OnInit{
>   ngOnInit(): void {
>     const map: maplibregl.Map = new maplibregl.Map({
>       container: 'map', // container id
>       style: {
>         version: 8,
>         sources: {
>           osm: {
>             type: "raster",
>             tiles: ["https://a.tile.openstreetmap.org/{z}/{x}/{y}.png"],
>             tileSize: 256,
>             attribution: "&copy; OpenStreetMap Contributors",
>             maxzoom: 19
>           }
>         },
>         glyphs: "https://fonts.openmaptiles.org/{fontstack}/{range}.pbf",
>         layers: [
>           {
>             id: "osm",
>             type: "raster",
>             source: "osm"
>           }
>         ]
>       },
>       center: [-8, 40], // starting position [lng, lat]
>       zoom: 15 // starting zoom
>     });
>     console.log('map', map)
>     map.addControl(new MeasuresControl(), "top-left");
> 
>   }
> }
> ````
> Then i added new control from maplibre-gl-measures and he partially works.
> I see button, he paint lines or polygon but i not see total area or length.
> ![map](https://github.com/user-attachments/assets/40e41a31-d1ea-47a5-ac2b-123514410d16)
> 
> The mistake is that maplibre-gl does not send request to glyphs at the specified address as in the example.
> 
> Please, help!
> What am I doing wrong?
> 
> I tried to use local glyphs. Also not worked.
> Unfortunately maplibre-gl not used glyphs, but when open example it`s working

Hi @SO-Matveev ! Could this be any config problem from the bundler? Are you using vite? Please take a look at issue #21 to see if the demo project that i've placed there helps you. Other than that, could you provide a sample project with this problem?

Cheers,

Jorge

**Reply by SO-Matveev** [reply, marked as answer]
Hello! Please  link to my project: https://github.com/SO-Matveev/my-map-app-by-maplipre
But i install your new version and it is works! Thank you!
I do not know what influenced for works, but it but it worked!
Also i still code in index.html ` var global = global || window`, because without it i give error about global is not defined

I sure, what your latest commits in project also influenced to work of the project! =)

**Reply by SO-Matveev** [reply]
Also, I noticed when used local glyphs, request is done, but response file is empty
![image](https://github.com/user-attachments/assets/c07ed6b6-8e7d-4db7-bc6a-3278d178464f)

Therefore i give error in console 
![image](https://github.com/user-attachments/assets/e283318e-9d1b-482d-b889-cf44a2d51d10)

**Reply by SO-Matveev** [reply]
I fix problem with use local glyphs in my project.
When you want use local glyphs and you paste folder glyphs with "Klokantech Noto Sans Bold" in folder assets, you need to make sure, that you config angular.json have settings in section "build" and section "serve":
````
            "assets": [
              "src/assets"
            ],
````
@jdsantos  I say thank you so much! You project is an important tool for working with the map!
If I can take the link for future use
[Demo maplibre-gl-measures on Angular](https://github.com/SO-Matveev/my-map-app-by-maplipre)
